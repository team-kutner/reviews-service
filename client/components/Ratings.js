import React from 'react';
import styled from 'styled-components'
import RatingsItem from './RatingsItem';

const RatingCountContainer = styled.div`
  padding-bottom: 32px;
`
const RatingsFlexContainer = styled.div`
  display: flex;
  align-items: stretch;

  justify-content: flex-start;
  flex-wrap: wrap;
  margin-bottom: 24px;
`
export default () => {
  return (
    <div>
      <RatingCountContainer>
        <span></span>
        (4 reviews)
      </RatingCountContainer>

      <RatingsFlexContainer>
        <RatingsItem/>
        <RatingsItem/>
        <RatingsItem/>
        <RatingsItem/>
        <RatingsItem/>
        <RatingsItem/>
      </RatingsFlexContainer>
    </div>
  );
};

