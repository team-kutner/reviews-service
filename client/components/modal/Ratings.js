import React from 'react';

const Ratings = (props) => {
  const {ratings, totalReviews} = props;

  return (
    <div>

    </div>
  );
}

export default Ratings;
