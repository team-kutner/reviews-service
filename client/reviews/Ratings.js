// import React from 'react';
const {React} = window;
// import styled, {css} from 'styled-components'
import calcOverallRating from '../utils/calcOverallRating';
import RatingsItem from './RatingsItem';
const {styled} = window
const {css} = styled

export default (props) => {
  const {ratings, totalReviews, modal=false } = props;


  return (
    <Container modal={modal}>
      <RatingCountContainer>
        <StarImage src={'https://i.imgur.com/U9jEbtL.png'} modal={modal}/>
        <TotalReviewsText modal={modal}>{calcOverallRating(ratings)} ({totalReviews} reviews)</TotalReviewsText>
      </RatingCountContainer>

      <RatingsFlexContainer modal={modal}>
        {Object.entries(ratings).map(([ratingName, ratingNumber]) => {


          return (
            <RatingsItem
            key={ratingName}
            ratingName={ratingName}
            ratingNumber={ratingNumber}
            modal={modal}
          />
          )
        })}

      </RatingsFlexContainer>
    </Container>
  );
};

const Container = styled.div`
  /* ${props => props.modal && css`
    width: 33%;
  `} */
`
const StarImage = styled.img`
  margin-right: 8px;
  ${props => props.modal && css`
    width: 25px;
  `}
`

const TotalReviewsText = styled.span`
  line-height: ${props => !props.modal ? '26px' : '36px'};
  color: ${props => props.theme.gray};
  font-weight: ${props => !props.modal ? props.theme.mdWeight : '800'};
  font-size: ${props => !props.modal ? props.theme.xlFont : '32px'};
`


const RatingCountContainer = styled.div`
  padding-bottom: 32px;

`
const RatingsFlexContainer = styled.div`
  display: flex;
  align-items: stretch;

  flex-direction: ${props => props.modal ? 'column' : 'row'};



  justify-content: flex-start;
  flex-wrap: wrap;
  margin-bottom: 24px;
`