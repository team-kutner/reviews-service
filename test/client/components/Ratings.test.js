import Enzyme, {shallow} from 'enzyme'

import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({adapter: new Adapter()})

import React from 'react'
import Ratings from '../../../client/reviews/Ratings'
import home10ApiResponse from '../../fixtures/home10ApiResponse'
const {reviewsWithComments, ratings} = home10ApiResponse

it('should render the ratings list and items', () => {
  const wrapper = shallow(<Ratings ratings={ratings} totalReviews={reviewsWithComments.length}/>)

  expect(wrapper).toMatchSnapshot()
})