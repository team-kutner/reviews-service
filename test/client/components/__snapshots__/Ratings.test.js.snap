// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the ratings list and items 1`] = `
<styled.div
  modal={false}
>
  <styled.div>
    <styled.img
      modal={false}
      src="https://i.imgur.com/U9jEbtL.png"
    />
    <styled.span
      modal={false}
    >
      3.9
       (
      100
       reviews)
    </styled.span>
  </styled.div>
  <styled.div
    modal={false}
  >
    <RatingsItem
      key="Cleanliness"
      modal={false}
      ratingName="Cleanliness"
      ratingNumber={3.8}
    />
    <RatingsItem
      key="Accuracy"
      modal={false}
      ratingName="Accuracy"
      ratingNumber={4.1}
    />
    <RatingsItem
      key="Communication"
      modal={false}
      ratingName="Communication"
      ratingNumber={4.1}
    />
    <RatingsItem
      key="Check-in"
      modal={false}
      ratingName="Check-in"
      ratingNumber={4}
    />
    <RatingsItem
      key="Value"
      modal={false}
      ratingName="Value"
      ratingNumber={3.9}
    />
    <RatingsItem
      key="Location"
      modal={false}
      ratingName="Location"
      ratingNumber={3.8}
    />
  </styled.div>
</styled.div>
`;
